tensorflow/compiler/mlir/hlo/lib/Dialect/mhlo/transforms/legalize_to_linalg.cc

```diff
@@ -912,6 +912,96 @@ class TransposeConverter
   }
 };
 
+/// Lowers mhlo.RealDynamicSliceOp to tensor.extract_slice and other
+/// arith/tensor dialect ops.
+class RealDynamicSliceConverter
+    : public OpConversionPattern<mhlo::RealDynamicSliceOp> {
+ public:
+  using OpConversionPattern<mhlo::RealDynamicSliceOp>::OpConversionPattern;
+
+  /// Computes size of a slice as :-
+  /// size = (limit - start)/stride
```

Comment:
I think this should be `(limit - 1 - start) / stride + 1)`. I hit this issue before and sent out a fix in slice op lowering.
```
Say that there are k elements in total, we have condition:
  start + (k - 1) * strides <= limit - 1
->
   k <= (limit - 1 - start) / strides + 1
```

An example is 

```
func @slice_with_strides2(%arg0: tensor<6xi32>) -> tensor<3xi32> {
  %0 = "mhlo.slice"(%arg0) {
    limit_indices = dense<5> : tensor<1xi64>,
    start_indices = dense<0> : tensor<1xi64>,
    strides = dense<2> : tensor<1xi64>
  } : (tensor<6xi32>) -> tensor<3xi32>
  return %0 : tensor<3xi32>
}
```

In this case, the total size is 3 (index 0, 2, 4), not `(5 - 0) / 2 = 2`.

Reply:
Shouldn't it be: 
```ceildiv((limit - start)/stride)```
?
This will yield `2` and not `3`.

Reply:
I think you are right on ceildiv, which is what this PR is doing. Please update the comment.

(btw, It will yield `3` not `2`?)

Reply:
Yes, it will yield `3`.
I've updated the comment.